revamp  the sidebar and implement the filter search

omotoshoayomikun opened the pull request:
revamp  the sidebar and implement the filter search

omotoshoayomikun committed:
revamp  the sidebar and implement the filter search

revamp  the sidebar and implement the filter search

omotoshoayomikun committed:
Filter Search Updated

omotoshoayomikun committed:
updated code

swoocn committed:
PR adjustment to fix vulnerabilities.

swoocn commented:
Hi @omotoshoayomikun - 
I’ve noticed some unusual behavior with the toggling of the filters. Sometimes when one filter is turned off, another one gets activated, and vice versa. My understanding is that there shouldn’t be any dependencies between the filters; each one should toggle independently without affecting others.

Additionally, I tested my store with the filters, turning both applicable filters off, but I can still see my store when I expected not to. It’s a Test Seller with a Trust Meter score of 100%. **[See screenshot below]**

![image](https://github.com/user-attachments/assets/ace93f59-6f2b-4623-8bb4-1eb38abcaf32)

omotoshoayomikun committed:
Merge branch 'dev' into feature/filter_search

omotoshoayomikun committed:
solve the unusual behavior with the toggling of the filters

swoocn committed:
Merge remote-tracking branch 'origin/dev' into feature/filter_search

swoocn committed:
Misc PR adjustments; integrate translation + cleanup.

swoocn commented:
FE changes look good after incorporating the translation, @omotoshoayomikun.  Great job! 👍

**[Current screenshot]**

![image](https://github.com/user-attachments/assets/1e86e773-6758-4f0e-a121-5d2a19271f9b)

Don't suppose you could make a couple UI adjustments?
1. Keep the 'Search Filters' subheader centered when toggled and the list is expanded.  Also, can you center the 'Personalization' subheader when it gets toggled as well?
2. It would be great if you could keep the filter options on the same line when the sidebar is displayed on smaller viewpoints.  Right now, the wording gets wrapped, which affects the display.  This will be especially important for translated versions.

Thanks!

omotoshoayomikun committed:
update on the filter search

1. Keep the 'Search Filters' and Personalization' subheader centered when toggled and the list is expanded.
2. keep the filter options on the same line when the sidebar is displayed on smaller viewpoints

omotoshoayomikun committed:
Merge branch 'feature/filter_search' of https://github.com/map-of-pi/map-of-pi-frontend-react into feature/filter_search

swoocn reviewed (approved):
Great job, @omotoshoayomikun! 😎👍 
Your FE PR is approved.
